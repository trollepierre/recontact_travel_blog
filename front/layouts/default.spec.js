import Vuex from 'vuex'
import Layout from './default.vue'

describe('default layout', () => {
  let localVue
  let store

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(Vuex)
    store = theme => new Vuex.Store({
      actions: {},
      state: { theme },
    })
  })

  it('should return default layout with aliceblue by default', () => {
    console.error = jest.fn()

    const wrapper = shallowMount(Layout, { localVue, store: store('light') })

    expect(wrapper).toMatchSnapshot()
  })

  it('should return default layout with dark', () => {
    console.error = jest.fn()

    const wrapper = shallowMount(Layout, { localVue, store: store('dark') })

    expect(wrapper.classes()).toContain('dark-mode')
  })

  it('should return default layout with new', () => {
    console.error = jest.fn()

    const wrapper = shallowMount(Layout, { localVue, store: store('new') })

    expect(wrapper.classes()).toContain('new-mode')
  })
})
