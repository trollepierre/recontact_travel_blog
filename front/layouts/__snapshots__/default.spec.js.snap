// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default layout should return default layout 1`] = `
<div id="app" class="aliceblue-mode">
  <nuxt></nuxt>
</div>
`;

exports[`default layout should return default layout with dark 1`] = `
<div id="app" class="dark-mode">
  <nuxt></nuxt>
</div>
`;
